jujugui/static/gui/src/app/components/user-profile/empty-user-profile.js

```diff
@@ -22,14 +22,55 @@ YUI.add('empty-user-profile', function() {
 
   juju.components.EmptyUserProfile = React.createClass({
     propTypes: {
-      staticURL: React.PropTypes.string
+      addNotification: React.PropTypes.func.isRequired,
+      env: React.PropTypes.object.isRequired,
+      hideConnectingMask: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object,
+      showConnectingMask: React.PropTypes.func.isRequired,
+      staticURL: React.PropTypes.string,
+      switchModel: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object
     },
 
+    getDefaultProps: function() {
+      return {
+        staticURL: ''
+      };
+    },
+
+    /**
+
+      A simply wrapper around the passed-in switchModel function; all the
+      wrapper does in pass an empty modelList into the provied switchModel.
+      After all, this is an EmptyUserProfile component; by definition we know
+      the modelList is empty.
+
+      @method switchModel
+      @param {String} uuid The model UUID.
+      @param {String} name The model name.
+      @param {Function} callback The function to be called once the model has
+        been switched and logged into. Takes the following parameters:
+        {Object} env The env that has been switched to.
+    */
+    switchModel: function(uuid, name, callback) {
+      this.props.switchModel(uuid, [], name, callback);
+    },
+
+
     render: function() {
-      var staticURL = this.props.staticURL || '';
```

Comment:
Again here, I prefer to not have the default prop value if it's only being used in a single location. I'll leave it up to you though.